FileManager/src/org/openintents/filemanager/bookmarks/BookmarksProvider.java

```diff
@@ -48,14 +53,28 @@
     private static final int DATABASE_VERSION = 2;
 
     private static class DatabaseHelper extends SQLiteOpenHelper {
+        private Context context;
 
         DatabaseHelper(Context context) {
             super(context, DATABASE_NAME, null, DATABASE_VERSION);
+            this.context = context;
         }
 
         @Override
         public void onCreate(SQLiteDatabase db) {
             db.execSQL(DATABASE_CREATE);
+            // Fill in bookmarks for secondary external storage on KitKat
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
+                for (File dir: KitKatExternalStorage.getExternalStoragePaths(context)) {
+                    ContentValues values = new ContentValues();
+                    values.put(BookmarksProvider.NAME, context.getString(R.string.external_storage) + " " + context.getString(R.string.read_only));
+                    values.put(BookmarksProvider.PATH, dir.getParentFile().getParentFile().getParentFile().getParentFile().getPath());
+                    db.insert(TB_NAME, "", values);
+                    values.put(BookmarksProvider.NAME, context.getString(R.string.external_storage));
+                    values.put(BookmarksProvider.PATH, dir.getPath());
+                    db.insert(TB_NAME, "", values);
+                }
```

Comment:
I prefer to have this as a button in the bookmarks view, instead of auto generated. External storage paths can change.